app/src/main/java/com/cops/iitbhu/previewer/BitmapYoutube.kt

```diff
@@ -0,0 +1,34 @@
+package com.cops.iitbhu.previewer
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.lifecycle.lifecycleScope
+import com.cops.iitbhu.previewer.databinding.ActivityBitmapYoutubeBinding
+import com.cops.iitbhu.previewer.lib.Previewer
+
+class BitmapYoutube : AppCompatActivity() {
+    private lateinit var binding: ActivityBitmapYoutubeBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityBitmapYoutubeBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.generateThumbnailForYoutube.setOnClickListener {
+            val youtubeLink = binding.youtubeLink.text.toString()
+            if (youtubeLink.isNotEmpty()) {
+                setThumbnail(youtubeLink)
+//                Previewer.setThumbnailFromYouTubeVideoUrl(youtubeLink, binding.img)
```

Comment:
Add a new button in the activity which will use this non-suspend version to directly set image to ImageView.